dart/lib/src/transport/http_transport.dart

```diff
@@ -19,19 +22,22 @@ class HttpTransport implements Transport {
 
   final RateLimiter _rateLimiter;
 
+  final ClientReportRecorder _clientReportRecorder;
```

Comment:
Maybe `ClientReportRecorder` should be a property of `SentryOptions`? cc @philipphofmann

Reply:
We don't put dependencies / objects to the options in ObjC. I guess you should follow your SDKs conventions and you can answer best how to follow these I think.

Reply:
We don't really have a central object/place to init dependencies AFAIK, but we rather worked with singletons for now, or other objects construct their dependencies internally, like in this PR.

Reply:
With the refactoring to not expose the new classes I moved the instance of `ClientReportRecorder ` to options. It is mainly used to access the instance in `Hub`, `SentryClient` and tests.